### PR TITLE
support dataframe in pytorch fit, pytorch evaluate, add pytorch predict

### PR DESCRIPTION
**Pull request by yangw1234**

### CODE REVIEW

File: pyzoo/zoo/orca/learn/tf2/estimator.py [hunk 1]
```diff
@@ -269,7 +219,7 @@ def zip_func(worker, this_shards_ref, that_shards_ref):
 
     def evaluate(self, data, batch_size=32, num_steps=None, verbose=1,
                  sample_weight=None, callbacks=None, data_config=None,
-                 feature_cols=None, label_cols=None):
+                 feature_cols=None, labels_cols=None):
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/tf2/estimator.py:222]: why `labels_cols`?

File: pyzoo/zoo/orca/learn/tf2/estimator.py [hunk 2]
```diff
@@ -214,26 +176,14 @@ def fit(self, data, epochs=1, batch_size=32, verbose=1,
...
-            schema = data.schema
-            numpy_rdd = data.rdd.map(lambda row: convert_row_to_numpy(row,
-                                                                      schema,
-                                                                      feature_cols,
-                                                                      label_cols))
-            shard_rdd = numpy_rdd.mapPartitions(lambda x: arrays2dict(x,
-                                                                      feature_cols,
-                                                                      label_cols))
-            data = SparkXShards(shard_rdd)
+        data, _ = maybe_dataframe_to_xshards(data, validation_data,
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/tf2/estimator.py:179]: why ignoring `validation_data` here?

File: pyzoo/zoo/orca/learn/pytorch/estimator.py [hunk 3]
```diff
@@ -90,7 +90,8 @@ def __init__(self,
                                              backend=backend,
                                              workers_per_node=workers_per_node)
 
-    def fit(self, data, epochs=1, batch_size=32, profile=False, reduce_results=True, info=None):
+    def fit(self, data, epochs=1, batch_size=32, profile=False, reduce_results=True, info=None,
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/estimator.py:93]: Need to support Spark Dataframes in `PyTorchSparkEstimator` too

File: pyzoo/zoo/orca/learn/pytorch/pytorch_ray_estimator.py [hunk 4]
```diff
@@ -182,12 +183,21 @@ def train(self,
...
+              info=None,
+              feature_cols=None,
+              labels_cols=None):
         """
         See the documentation in
         'zoo.orca.learn.pytorch.estimator.PyTorchRayEstimatorWrapper.fit'.
         """
         from zoo.orca.data import SparkXShards
+
+        data, _ = maybe_dataframe_to_xshards(data,
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/pytorch_ray_estimator.py:195]: why ignore `validation_data` here?

File: pyzoo/zoo/orca/learn/tf2/estimator.py (hunk 1 above)
Comment (yangw1234) [reply pyzoo/zoo/orca/learn/tf2/estimator.py:222]: I do think `label_cols` make more sense, however other estimators all call it `labels_cols`. How about I make another pr to change them all to `label_cols`

File: pyzoo/zoo/orca/learn/pytorch/pytorch_ray_estimator.py (hunk 4 above)
Comment (yangw1234) [reply pyzoo/zoo/orca/learn/pytorch/pytorch_ray_estimator.py:195]: this estimator has not support `validation_data` yet

File: pyzoo/zoo/orca/learn/tf2/estimator.py (hunk 2 above)
Comment (yangw1234) [reply pyzoo/zoo/orca/learn/tf2/estimator.py]: fixed

File: pyzoo/zoo/orca/learn/tf2/estimator.py (hunk 1 above)
Comment (jason-dai) [reply pyzoo/zoo/orca/learn/tf2/estimator.py:222]: > I do think `label_cols` makes more sense, however other estimators all call it `labels_cols`. How about I make another pr to change them all to `label_cols` ?

ok

File: pyzoo/zoo/orca/learn/pytorch/pytorch_ray_estimator.py [hunk 5]
```diff
@@ -302,6 +299,41 @@ def transform_func(worker, shards_ref):
...
 
+    def predict(self,
+                data,
+                batch_size=32,
+                profile=False,
+                feature_cols=None,
+                labels_cols=None):
+        """
+        See the documentation in
+        'zoo.orca.learn.pytorch.estimator.PyTorchRayEstimatorWrapper.evaluate'.
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/pytorch_ray_estimator.py:310]: What's this for?

Comment (yangw1234) [reply pyzoo/zoo/orca/learn/pytorch/pytorch_ray_estimator.py]: fixed wrong comments

File: pyzoo/zoo/orca/learn/pytorch/estimator.py (hunk 3 above)
Comment (yangw1234) [reply pyzoo/zoo/orca/learn/pytorch/estimator.py:93]: will add it in the following pr

**Comment by yangw1234** [comment]
http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/4932/ jenkins passed except for a few unrelated failures

**Pull request merged**